### PR TITLE
Point changes to inbox_type and inbox_guidelines attributes to users_settings table

### PR DESCRIPTION
**Pull request by msarit**
## Description
This PR point changes to `inbox_type` and `inbox_guidelines` within the `Extensions` tab to the `users_settings` table. This PR shall be merged into [PR 13904](https://github.com/forem/forem/pull/13904).

## TO-DO list for this PR
- [x] Update relevant attributes in the `customization` and `profile` tabs to use the new `users_settings` table 
- [x] Go through the TO-DO's on the application 
- [x] Check for other instances where these fields are being used

## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## QA Instructions, Screenshots, Recordings
1. Sign into `localhost`; in rails console, pull up the same user that you are signed in as.
2. In rails console, run `user.setting` and take note the values for `inbox_type` and `inbox_guidelines`. Also run `user.inbox-type` and `user.inbox_guidelines` and note that the values match those in the `notification_setting` record. 
3. In the app, navigate to `localhost:xxxx/settings/extensions`; toggle the Inbox Privacy setting, and enter some content in the Open inbox guidelines/instructions field. Click Save Connect Settings. You should be redirected to the Profile tab.
4.  In the console, reload your user (`user.reload`) and re-run `user.setting`, `user.inbox_type` and `user.inbox-guidelines`. Now, the `setting` values should differ from the `user.` values, as the `users` record was left unchanged, but the `settings` record was changed accordingly. 

## Added tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests
- [X] Not yet; will write after merging into [PR 13904](https://github.com/forem/forem/pull/13904)

## [Forem core team only] How will this change be communicated?

- [X] I will share this change internally with the appropriate teams

### CODE REVIEW

**Review by jgaskins — APPROVED** [review]
This looks great! Just one minor nitpick that I got completely [nerd-sniped](https://xkcd.com/356/) by. :joy: It's completely nonblocking, but I did want to point it out because it's a pretty interesting quirk of the collective interactions between Rails controllers, HTML, HTTP, ActiveRecord, and Postgres.

File: spec/requests/user/user_settings_spec.rb [hunk 1]
```diff
@@ -257,8 +257,8 @@
       expect(user.setting.display_announcements).to be(true)
 
       expect do
-        put "/users/#{user.id}", params: { user: { tab: "misc", display_announcements: 0 } }
-      end.to change { user.reload.display_announcements }.from(true).to(false)
+        patch users_setting_path(user.setting.id), params: { users_setting: { display_announcements: false } }
```
[the remark this answers is not in the document]
Comment (msarit) [reply spec/requests/user/user_settings_spec.rb:260]: (le sigh ...) What would I do without you, @jgaskins 🙏🏾

[the remark this answers is not in the document]
Comment (jgaskins) [reply spec/requests/user/user_settings_spec.rb:260]: You'd have a lot less to read, that's for sure :joy:

[the remark this answers is not in the document]
Comment (rhymes) [reply spec/requests/user/user_settings_spec.rb]: Great explanation!

**Pull request merged**